codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/OperationGenerator.java

```diff
@@ -92,8 +96,12 @@ public void run() {
                         writer.write("if err := fn(stack); err != nil { return nil, err }");
                     });
 
-                    constructHandler();
+                    if (protocolGenerator != null) {
```

Comment:
This check will need to go into the function itself, as the function is going to be used to add other middleware. Probably not a big deal unless that other PR gets merged first.

Reply:
^^ Sorry this is based on a PR by @skotambkar that is old and will be removed once his middleware PRs are merged